doc/sphinx/proof-engine/ltac.rst

```diff
@@ -582,10 +582,35 @@ Failing
       there are no goals left. Be careful however if Coq terms have to be
       printed as part of the failure: term construction always forces the
       tactic into the goals, meaning that if there are no goals when it is
-      evaluated, a tactic call like :n:`let x:=H in fail 0 x` will succeed.
+      evaluated, a tactic call like :n:`let x := H in fail 0 x` will succeed.
 
    .. exn:: Tactic Failure message (level @num).
 
+   .. exn:: No such goal.
+      :name: No such goal. (fail)
+
+   .. example::
+
+      .. coqtop:: all
+
+         Goal True.
+         Proof. fail. Abort.
+
+         Goal True.
+         Proof. trivial; fail. Qed.
+
+         Goal True.
+         Proof. trivial. fail. Abort.
```

Comment:
This example doesn't seem very useful given that there is nothing specific to `fail` here. Unless you add `all:` in front of `fail`.